src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java

```diff
@@ -493,6 +501,23 @@ private void init(long recordOffset) {
       // recover will only check pipe status and create pipe if it does not exist.
       recover(pipeCreation);
 
+      // when exactly_once is enabled,fetch clientSequencer and offsetPersistedInSnowflake
+      if (ingestionDeliveryGuarantee
```

Comment:
lets extract this out to a new method so that this init method is not noisy. 
Also since it calls only during start, we should log the offsetToken and clientSequencer.

Reply:
Sounds good